solutions/andy_ashlee_kotlin/main/kotlin/co/jfactory/yorktdd/Words.kt

```diff
@@ -16,9 +16,7 @@ fun findSteps(start: String, finish: String, words : List<String>): List<String>
 
     while (queue.isNotEmpty()){
         val previousWords = queue.removeAt(0)
-        val nextWords = findCloseWords(previousWords.last(), words).
-
-
+        val nextWords = findCloseWords(previousWords.last(), words).filterNot { it in previousWords }
```

Comment:
Ah, this is the same as the find I did; but quite elegant just tagging it on the end....

Reply:
I really like Kotlin's implementation of Collection functions.  They are quite similar to those introduced in Java 8 but the syntax for the lambda functions is much cleaner.